### PR TITLE
zyl upload his homework

### PR DESCRIPTION
**Pull request by triode-zyl**
happy day!

### CODE REVIEW

File: week_3/README.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+HomeWork3
```
Comment (Menci) [inline week_3/README.md:1]: You shouldn't add this file.

File: week_3/2018202196ZYL/src/crawler.h [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+#define CRAWLER_H
+
+#include<queue>
+#include<set>
+#include<string>
+class crawler
+{
+	private:
+		std::set <std::string> str_set;
+		std::pair<std::set<std::string>::iterator,bool>ret;
```
Comment (Menci) [inline week_3/2018202196ZYL/src/crawler.h:11]: Why use a `std::pair` rather than two variables?

File: week_3/2018202196ZYL/src/crawler.cpp [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+		ans++;
+		//const char* str_que=str_queue.front().data();
+		//sprintf(wget_str,"wget %s -O ./home/%d.html -T 20 -q",str_que,ans);
+		wget_str="wget \""+str_queue.front()+"\" --output-document=./home/"+std::to_string(ans)+".html -t 3 -T 15 -q";
+		system(wget_str.c_str());
+		std::cout<<ans<<' '<<str_queue.front()<<std::endl;
+		str_queue.pop();
+		//sprintf(str,"./home/%d.html",ans);
+		str="./home/"+std::to_string(ans)+".html";
+		std::ifstream cur_url (str.c_str());
```
Comment (Menci) [inline week_3/2018202196ZYL/src/crawler.cpp:28]: Indent your code with either spaces or tabs, not BOTH spaces AND tabs. Tabs' length may be different in everywhere.

File: week_3/README.md (hunk 1 above)
Comment (WhiteCatFly) [reply week_3/README.md:1]: I fixed that. Hope everyone can follow the rules

**Pull request merged**